.github/workflows/nix.yml

```diff
@@ -0,0 +1,105 @@
+name: Nix CI
+
+on:
+  pull_request:
+    paths:
+      - '.github/workflows/nix.yml'
+      - 'nix/**'
+      - 'src/**'
+      - 'Cargo.*'
+      - 'flake.*'
+  push:
+    branches:
+      - main
+    paths:
+      - '.github/workflows/nix.yml'
+      - 'nix/**'
+      - 'src/**'
+      - 'Cargo.*'
+      - 'flake.*'
+  workflow_dispatch:
+
+permissions:
+  contents: read
+
+concurrency:
+  group: ${{ github.sha }}
+  cancel-in-progress: true
```

Comment:
ran into issues with this in Mamba. You may want to consider making jobs wait instead of canceling, at least on the main branch, to ensure each commit is cached. You can look here to see what I am doing currently:
https://github.com/input-output-hk/mamba-world/blob/a6e47b05dcb389051a8ce213a3e86270810c551e/.github/workflows/std.yml